src/libs/ajax/User.ts

```diff
@@ -74,17 +94,44 @@ export interface SetTerraUserProfileRequest {
   interestInTerra?: string;
 }
 
-export const makeSetUserProfileRequest = (terraUserProfile: TerraUserProfile): SetTerraUserProfileRequest => {
+export const generateAPIBodyForCreateUserProfile = (
+  request: CreateTerraUserProfileRequest
+): OrchestrationUpsertTerraUserProfileRequest => {
+  return generateAPIBodyForUpdateUserProfile(request);
+};
+
+export const generateAPIBodyForUpdateUserProfile = (
+  request: UpdateTerraUserProfileRequest
+): OrchestrationUpsertTerraUserProfileRequest => {
   return {
-    firstName: terraUserProfile.firstName ?? 'N/A',
-    lastName: terraUserProfile.lastName ?? 'N/A',
-    title: terraUserProfile.title ?? 'N/A',
-    institute: terraUserProfile.institute ?? 'N/A',
-    programLocationCity: terraUserProfile.programLocationCity ?? 'N/A',
-    programLocationState: terraUserProfile.programLocationState ?? 'N/A',
-    programLocationCountry: terraUserProfile.programLocationCountry ?? 'N/A',
-    department: terraUserProfile.department,
-    interestInTerra: terraUserProfile.interestInTerra,
+    // TODO: this is the logic to look at when writing the tests
+    // first name and last name are REQUIRED for this request, and they are populated from the oidc token claims,
+    // so they should not be undefined
+    firstName: request.firstName!,
+    lastName: request.lastName!,
+
+    // contact email is NOT REQUIRED for this request, but it is populated from the oidc token claims,
+    // so it should not be undefined
+    contactEmail: request.contactEmail!,
+
+    // title and institute are REQUIRED for this request, but they do not necessarily
+    // get set during registration
+    title: !_.isEmpty(request.title) ? request.title : 'N/A',
```

Comment:
Nit: suggest switching these conditions around to avoid the double negative of `!_.isEmpty`:

```suggestion
    title: _.isEmpty(request.title) ? 'N/A' : request.title,
```